scarb/src/ops/metadata.rs

```diff
@@ -125,10 +126,13 @@ fn collect_package_metadata(package: &Package) -> m::PackageMetadata {
         .build()
         .unwrap();
 
+    let edition = to_variant_name(&package.manifest.edition).unwrap();
```

Comment:
hmm, why? what's the difference from just serializing the edition value?

Reply:
```
#[test]
fn test() {
    let e = Edition::V2023_10;
    assert_eq!(to_variant_name(&e).unwrap(), "2023_10");
    assert_eq!(serde_json::to_string(&e).unwrap(), "2023_10");
}
```
The second assert fails with 
```
assertion `left == right` failed
  left: "\"2023_10\""
 right: "2023_10"
```

Reply:
I think we could avoid this dependency by serializing to `serde_json::Value` and then deserializing as `String`. This way we could also better handle cases `Edition` serialization produces something unexpected—current code would produce broken output, while with my suggestion, we would early panic instead.